plugins/source/metrics.go

```diff
@@ -123,3 +124,42 @@ func (s *Metrics) TotalResourcesAtomic() uint64 {
 	}
 	return total
 }
+
+func (s *Metrics) MarkStart(table *schema.Table, clientID string) {
+	s.TableClient[table.Name][clientID].StartTime = time.Now()
+}
+
+// If the table is a top-level table, we need to mark all of its descendents as 'done' as well.
+// This is because, when a top-level table is empty (no resources), its descendants are never actually
+// synced.
+func (s *Metrics) MarkEnd(table *schema.Table, clientID string) {
+	if table.Parent == nil {
+		s.markEndRecursive(table, clientID)
+	} else {
+		s.TableClient[table.Name][clientID].EndTime = time.Now()
+	}
+}
+
+func (s *Metrics) markEndRecursive(table *schema.Table, clientID string) {
+	s.TableClient[table.Name][clientID].EndTime = time.Now()
+	for _, relation := range table.Relations {
+		s.markEndRecursive(relation, clientID)
+	}
+}
+
+func (s *Metrics) InProgressTables() []string {
```

Comment:
I would say it's better to print "tables in progress longer that X time", otherwise we will print tables that are short to sync along with tables that are long to sync

Reply:
I think we can leave this optimization out of the first version of this PR, because most tables start at the same time in almost all cases.
Also, the logggin code makes sure to only log.

This PR tries to help with the annoying pattern of cloudquery where one (or couple) of tables take 10X or more time than all the rest - and this should be way more visible with this change.